### PR TITLE
firebase config

### PR DESCRIPTION
**Pull request by Shoxer90**

### CODE REVIEW

File: src/Components/Contacts/ContactItem.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+const ContactItem = () => {
+    return (
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/ContactItem.js:4]: If your component return just dummy HTML and there is no methods or other functional except  HTML please use this format.

const ContactItem = () => ([YOUR_HTML])

File: src/Components/Contacts/ContactItem.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+        <div>
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/ContactItem.js:5]: If your component has no children the don't use opening and closing tags just write (<div />)

File: src/Components/Contacts/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React, { memo } from "react";
+import ContactItem from "./ContactItem";
+
+const Contacts = ({contacts}) => {
+    return (
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/index.js:5]: Same here (remove the "return" operator)

File: src/Components/Contacts/index.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        contacts.map((contact) => (
+            <div key={contact.id}>
+                <ContactItem {...contact}/>
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/index.js:8]: If your component is closing itself and it contains just one line the closing "/" need to have one space before it, like

<ContactItem {...contact} />

File: src/Components/Contacts/ContactItem.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+        </div>
+    );
+};
+
+export default ContactItem;
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/ContactItem.js:11]: Please memoize this component

File: src/Components/Contacts/index.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, { memo } from "react";
+import ContactItem from "./ContactItem";
+
+const Contacts = ({contacts}) => {
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/index.js:4]: If your object contains only one line please set spaces after opening curly bracket "{" and before  closing curly bracket "}"

File: src/Components/Contacts/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React, { memo } from "react";
+import ContactItem from "./ContactItem";
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/index.js:2]: Please separate component imports from package imports

File: src/Components/Contacts/index.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+
+const Contacts = ({contacts}) => {
+    return (
+        contacts.map((contact) => (
+            <div key={contact.id}>
+                <ContactItem {...contact}/>
+            </div>
+        ))
```
Comment (Suren-Sargsyan14) [inline src/Components/Contacts/index.js:10]: please set ; in the end of this line

File: src/Components/Header/index.js [hunk 9]
```diff
@@ -11,7 +11,7 @@ const Header = ({openComponent}) => {
             <Button 
                 h="35px"
```
Comment (Suren-Sargsyan14) [inline src/Components/Header/index.js:12]: I think "h" and "w" is not a good names for props, I understand that they mean width and height but you need to change these names of props

File: src/Components/Header/index.js [hunk 10]
```diff
@@ -11,7 +11,7 @@ const Header = ({openComponent}) => {
...
                 w="100px"
-                name={"+"} 
+                name={"Add new contact"}
```
Comment (Suren-Sargsyan14) [inline src/Components/Header/index.js:14]: I saw that in the top for w and h props you used propName="propValue" syntax, if you are using some syntax please keep it everywhere. Here you wrote propName={PropValue}, but the prop value is a dummy string

File: src/Components/NewContact/index.js [hunk 11]
```diff
@@ -1,11 +1,104 @@
...
+
+    useEffect(() => {
+        setContacts(state)
+        alert("Contact is ready")
+    }, [state.avatar]);
+
+    const handleSetContact = (e) => {
+        e.preventDefault();
+        const imageRef = ref(fireStorage, `contactAvatars/${e.target[4].files[0].name}`);
+            uploadBytes( imageRef, e.target[4].files[0] ).then(() => {
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.js:35]: Don't set spaces before and after "(" and ")"

File: src/Services/getContacts.js [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { collection, query, getDocs } from "firebase/firestore";
+import { fireStore } from "../Config/firebaseInit";
+
+const getContacts = async() => {
```
Comment (Suren-Sargsyan14) [inline src/Services/getContacts.js:4]: please set space after async

File: src/Services/getContacts.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    const q = query(collection(fireStore, "contacts"));
+    const querySnapshot = await getDocs(q);
+    querySnapshot.forEach((doc) => {
+     console.log(doc.id, " => ", doc.data());
```
Comment (Suren-Sargsyan14) [inline src/Services/getContacts.js:8]: Please remove unnecessary console

File: src/Services/getContacts.js [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import { collection, query, getDocs } from "firebase/firestore";
+import { fireStore } from "../Config/firebaseInit";
+
+const getContacts = async() => {
+    const q = query(collection(fireStore, "contacts"));
```
Comment (Suren-Sargsyan14) [inline src/Services/getContacts.js:5]: Please keep collection name in constant variable

File: src/Services/getContacts.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+    const querySnapshot = await getDocs(q);
+    querySnapshot.forEach((doc) => {
+     console.log(doc.id, " => ", doc.data());
+
```
Comment (Suren-Sargsyan14) [inline src/Services/getContacts.js:9]: Please remove unnecessary line

File: src/Services/setContacts.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+import { doc, setDoc } from "firebase/firestore";
+import { fireStore } from "../Config/firebaseInit";
+
+import { v4 as uuidv4 } from 'uuid';
+
+const setContacts = async(form) => {
```
Comment (Suren-Sargsyan14) [inline src/Services/setContacts.js:6]: please set space after async

File: src/Services/setContacts.js [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+    await setDoc(doc(fireStore, "contacts", form.email), {
```
Comment (Suren-Sargsyan14) [inline src/Services/setContacts.js:7]: in this function the async await is unnecessary, you can just return the function which you called inside this function

File: src/Components/NewContact/index.js [hunk 18]
```diff
@@ -1,11 +1,104 @@
...
+        setState({
+            ...state,
+            [e.target.name]: e.target.value,
+        });
+    };
+
+    useEffect(() => {
+        setContacts(state)
+        alert("Contact is ready")
+    }, [state.avatar]);
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.js:30]: You can set a contact inside handleSetContact function, so please remove unnecessary useEffect

File: src/Components/NewContact/index.js [hunk 19]
```diff
@@ -1,11 +1,104 @@
...
+
+    const handleSetContact = (e) => {
+        e.preventDefault();
+        const imageRef = ref(fireStorage, `contactAvatars/${e.target[4].files[0].name}`);
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.js:34]: Also as I remember I wanted to upload multiple photos, but as I can see you wrote code just for uploading one image

File: src/Components/NewContact/index.js [hunk 20]
```diff
@@ -1,11 +1,104 @@
...
+        <form onSubmit={handleSetContact} className={styles.newContact}>
+            <label className={styles.label}>
+                Name: 
+                <input
+                    type="text"
+                    name="firstName"
+                    value={state.firstName}
+                    onChange={handleInputChange}
+                />
+            </label>
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.js:63]: This format is duplicated. I mean there is many inputs and their difference is just in some props. You can create a constant array and fill it with objects like this one
[
  {
    name: 'firstName',
    type: 'text',
    label: 'Name: ',
  },
]
and then write a map() and render all these objects.
I mean all these inputs beside avatar

File: src/Components/NewContact/index.js [hunk 21]
```diff
@@ -1,11 +1,104 @@
...
+            <label>
+                Get photo 
+                <input 
+                    type="file" 
+                    name="avatar" 
+                />
+            </label>
+            <button type="submit">add contact</button>
+            <button>go back</button>
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.js:99]: You can set type of this button reset, of course if you want

File: src/Components/NewContact/index.module.scss [hunk 22]
```diff
@@ -0,0 +1,17 @@
+.newContact{
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.module.scss:1]: please set a space before curly bracket {

File: src/Components/NewContact/index.module.scss [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+    border: solid black 1px;
+    border-radius: 10px;
+    padding: 20px;
+    text-align: center;
+    height: 500px;
+    width: auto;
+    label{
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.module.scss:8]: same here

File: src/Components/NewContact/index.module.scss [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+        display: flex;
+        margin: 10px;
+        input{
+            width: 80%;
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.module.scss:12]: same here

File: src/Components/NewContact/index.module.scss [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+        }
+
```
Comment (Suren-Sargsyan14) [inline src/Components/NewContact/index.module.scss:14]: please remove unnecessary line

File: src/Components/Search/index.js [hunk 26]
```diff
@@ -1,19 +1,25 @@
...
         <div className={styles.search}>
-            <form onClick={(e)=>handleSearch}>
-                <input className={styles.input} type="text" placeholder="Get contact from your contact list..." />
-                <button type="submit" >Search</button>
+            <form onSubmit={(e)=>handleSearch(e)}>
+                <input 
+                    className={styles.input}
+                    type="text" 
+                    placeholder="Get contact from your contact list..."
+                    onChange={(e)=>setInputs(e.target.value)}
```
Comment (Suren-Sargsyan14) [inline src/Components/Search/index.js:20]: please set spaces before and after => (arrow)

File: src/Components/Search/index.js [hunk 27]
```diff
@@ -1,19 +1,25 @@
...
-        filterContacts();
+        filterContacts(inputs);
     };
 
     return (
         <div className={styles.search}>
-            <form onClick={(e)=>handleSearch}>
-                <input className={styles.input} type="text" placeholder="Get contact from your contact list..." />
-                <button type="submit" >Search</button>
+            <form onSubmit={(e)=>handleSearch(e)}>
```
Comment (Suren-Sargsyan14) [inline src/Components/Search/index.js:15]: please set spaces before and after => (arrow)

also you can write here like this  `<form onSubmit={handleSearch}>`

File: src/index.js [hunk 28]
```diff
@@ -6,8 +6,4 @@ import App from './App';
 import './index.css';
 
 const root = ReactDOM.createRoot(document.getElementById('root'));
-root.render (
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
-);
```
Comment (Suren-Sargsyan14) [inline src/index.js:13]: Why did you remove the React.StrictMode ? Please bring it back

File: src/Components/index.js [hunk 29]
```diff
@@ -1,23 +1,35 @@
...
+        console.log(tag)
+        getContacts()
     };
 
     return (
         <div>
             <Header openComponent={openComponent}/>
             <Search filterContacts={filterContacts}/>
+            {openNewContact && <NewContact/>}
+            {contacts && <Contacts {...contacts}/>}
```
Comment (Suren-Sargsyan14) [inline src/Components/index.js:32]: please add a space before `/>`

File: src/Components/index.js [hunk 30]
```diff
@@ -1,23 +1,35 @@
...
     const filterContacts = (tag) => {
-        console.log("filter")
+        console.log(tag)
+        getContacts()
     };
 
     return (
         <div>
             <Header openComponent={openComponent}/>
             <Search filterContacts={filterContacts}/>
```
Comment (Suren-Sargsyan14) [inline src/Components/index.js:30]: please add a space before `/>`

File: src/Components/index.js [hunk 31]
```diff
@@ -1,23 +1,35 @@
...
+       buildContacts();
+    },[]);
 
     const openComponent = () => {
         setStatusNewContact(!openNewContact)
     };
 
     const filterContacts = (tag) => {
-        console.log("filter")
+        console.log(tag)
```
Comment (Suren-Sargsyan14) [inline src/Components/index.js:23]: Please remove unnecessary console

File: src/Components/index.js [hunk 32]
```diff
@@ -1,23 +1,35 @@
...
+import NewContact from "./NewContact";
 import Search from "./Search";
 
 const Component = () => {
     const [openNewContact, setStatusNewContact] = useState(false);
+    const [contacts, buildContacts] = useState();
+
+    useEffect(() => {
+       buildContacts();
+    },[]);
```
Comment (Suren-Sargsyan14) [inline src/Components/index.js:16]: I don't know why did you write this, but please remove this useEffect, because it's unnecessary

**Review by Suren-Sargsyan14 — CHANGES_REQUESTED** [review]

File: src/Components/Contacts/ContactItem.js (hunk 1 above)
Comment (Shoxer90) [reply src/Components/Contacts/ContactItem.js:4]: I just change this component on the next branches,any way i fix a few components with the same issue
